src/storage.rs

```diff
@@ -68,26 +68,49 @@ impl Default for AccountBasic {
     }
 }
 
-/// EVM Code, currently mapping to REVM's [ByteCode::LegacyAnalyzed].
-// TODO: Support raw legacy & EOF
 #[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
-pub struct EvmCode {
-    /// Bytecode with 32 zero bytes padding
+pub struct LegacyAnalyzedCode {
+    /// Bytecode with 32 zero bytes padding.
     bytecode: Bytes,
-    /// Original bytes length
+    /// Original bytes length.
     original_len: usize,
     /// Jump table.
     jump_table: Arc<BitVec<u8>>,
 }
 
+#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
+pub struct Eip7702Code {
+    /// EOA which which will inherit the bytecode.
+    delegated_address: Address,
+    /// Version of the bytecode.
+    version: u8,
+    /// Raw bytecode encoding MAGIC, version and delegated address.
+    raw: Bytes,
```

Comment:
I don't think we need this for "storage". We can always re-construct it at runtime for execution (like at `revm` level) from `address` and `version`.

Reply:
Good callout. I've removed `raw` and kept `version` so we can preserve it. This has naturally made me choose option 2. from above.